shell/models/__tests__/catalog.cattle.io.app.test.ts

```diff
@@ -0,0 +1,80 @@
+import CatalogApp from '@shell/models/catalog.cattle.io.app';
+
+describe('class CatalogApp', () => {
+  describe('should return upgradeAvailable for this chart', () => {
+    describe('as false to be excluded from some logic if', () => {
+      const expectation = false;
+
+      it('is fleet', () => {
+        const catalogApp = new CatalogApp({ spec: { chart: { metadata: { annotations: { 'fleet.cattle.io/bundle-id': 'anything' } } } } });
+
+        expect(catalogApp.upgradeAvailable).toStrictEqual(expectation);
+      });
+
+      it('is managed,', () => {
+        const catalogApp = new CatalogApp({ spec: { chart: { metadata: { annotations: { 'catalog.cattle.io/managed': 'anything' } } } } });
+
+        expect(catalogApp.upgradeAvailable).toStrictEqual(expectation);
+      });
+    });
+
+    describe('as latest version', () => {
+      it('excluding pre-releases', () => {
+        const version = '1';
+        const catalogApp = new CatalogApp({});
+
+        catalogApp.spec = { chart: { metadata: { version: '0' } } };
+
+        jest.spyOn(catalogApp, '$rootGetters', 'get').mockReturnValue({
+          'catalog/chart': () => ({ versions: [{ version }] }),
+          currentCluster:  jest.fn(),
+          'prefs/get':     () => false
+        });
+
+        expect(catalogApp.upgradeAvailable).toStrictEqual(version);
+      });
+    });
+
+    describe('as null', () => {
+      const expectation = null;
+
+      it('if no chart', () => {
+        const catalogApp = new CatalogApp({});
+
+        jest.spyOn(catalogApp, '$rootGetters', 'get').mockReturnValue({
+          'catalog/chart': () => [],
+          currentCluster:  jest.fn(),
+          'prefs/get':     () => false
+        });
+
+        expect(catalogApp.upgradeAvailable).toStrictEqual(expectation);
+      });
+
+      it('if no version is available', () => {
```

Comment:
this is the same as above (should it include `catalogApp.spec = { chart: { metadata: {  } } };`?)

Reply:
No it's not necessary for computation.

Reply:
to check explicitly for a missing version it would need to get passed the check for a missing chart and spec. so by providing the object above the test should hit the code that it's aiming to check. 

as it stands the test is the same as for `if no chart` and doesn't exercise anything different

Reply:
but the test is passing, otherwise we are missing one and I did not realize it.

Reply:
the two tests `if no chart` and `if no version is available` are exactly the same. both tests should provide and env such that only the condition referenced in the test title is exercised